src/DataStructures/Tensor/Expressions/Evaluate.hpp

```diff
@@ -256,6 +258,65 @@ void evaluate_impl(
     }
   }
 }
+
+/*!
+ * \ingroup TensorExpressionsGroup
+ * \brief Assign a `double` value to components of the LHS tensor
+ *
+ * \details This is for internal use only and should never be directly called.
+ * See `tenex::evaluate` and use it, instead.
+ *
+ * \note `LhsTensorIndices` must be passed by reference because non-type
+ * template parameters cannot be class types until C++20.
+ *
+ * @tparam LhsTensorIndices the `TensorIndex`s of the `Tensor` on the LHS of the
+ * tensor expression, e.g. `ti::a`, `ti::b`, `ti::c`
+ * @param lhs_tensor pointer to the resultant LHS `Tensor` to fill
+ * @param rhs_value the RHS value to assigned
+ */
+template <auto&... LhsTensorIndices, typename X, typename LhsSymmetry,
+          typename LhsIndexList>
+void evaluate_impl(
+    const gsl::not_null<Tensor<X, LhsSymmetry, LhsIndexList>*> lhs_tensor,
+    const double rhs_value) {
+  constexpr size_t num_lhs_indices = sizeof...(LhsTensorIndices);
+
+  using lhs_tensorindex_list =
+      tmpl::list<std::decay_t<decltype(LhsTensorIndices)>...>;
+
+  static_assert(
+      tensorindex_list_is_valid<lhs_tensorindex_list>::value,
+      "Cannot assign a tensor expression to a LHS tensor with a repeated "
+      "generic index, e.g. evaluate<ti::a, ti::a>. (Note that the concrete "
+      "time indices (ti::T and ti::t) can be repeated.)");
+  static_assert(
+      not contains_indices_to_contract<num_lhs_indices>(
+          {{std::decay_t<decltype(LhsTensorIndices)>::value...}}),
+      "Cannot assign a tensor expression to a LHS tensor with generic "
+      "indices that would be contracted, e.g. evaluate<ti::A, ti::a>.");
+
+  // positions of indices in LHS tensor where generic spatial indices are used
+  // for spacetime indices
+  constexpr auto lhs_spatial_spacetime_index_positions =
+      get_spatial_spacetime_index_positions<LhsIndexList,
+                                            lhs_tensorindex_list>();
+
+  // positions of indices in LHS tensor where concrete time indices are used
+  constexpr auto lhs_time_index_positions =
+      get_time_index_positions<lhs_tensorindex_list>();
+
+  using lhs_tensor_type = typename std::decay_t<decltype(*lhs_tensor)>;
+
+  for (size_t i = 0; i < lhs_tensor_type::size(); i++) {
+    auto lhs_multi_index =
```

Comment:
I don't understand why you need any multi-index. the storage index is guaranteed to be unique. I'm not sure why you need an impl either, can't this just be:

```cpp
for (auto& component : *tensor) {
  component = rhs_value;
}
```
in the non-impl?